File: 0303-range-sum-query-immutable/0303-range-sum-query-immutable.js
/**
 * @param {number[]} nums
 */
var NumArray = function(nums) {
    this.prefix = [0];
    for (let i = 1; i < nums.length + 1; i++) {
        this.prefix[i] = this.prefix[i - 1] + nums[i - 1]; 
    }
};

/** 
 * @param {number} left 
 * @param {number} right
 * @return {number}
 */
NumArray.prototype.sumRange = function(left, right) {
    return this.prefix[right + 1] - this.prefix[left];
};

/** 
 * Your NumArray object will be instantiated and called as such:
 * var obj = new NumArray(nums)
 * var param_1 = obj.sumRange(left,right)
 */

File: 0303-range-sum-query-immutable/0303-range-sum-query-immutable.py
class NumArray:
    def __init__(self, nums):
        self.prefix_sums = [0] * (len(nums) + 1)

        for i in range(1, len(nums) + 1):
            self.prefix_sums[i] = self.prefix_sums[i-1] + nums[i-1]

    def sumRange(self, left, right):
        return self.prefix_sums[right+1] - self.prefix_sums[left]        


# Your NumArray object will be instantiated and called as such:
# obj = NumArray(nums)
# param_1 = obj.sumRange(left,right)